api.yaml

```diff
@@ -2,51 +2,94 @@ openapi: 3.0.0
 servers:
   # Added by API Auto Mocking Plugin
   - description: SwaggerHub API Auto Mocking
-    url: https://virtserver.swaggerhub.com/Noel/TBG-Donations/1.0.10
+    url: https://virtserver.swaggerhub.com/Noel/TBG-Donations/1.0.12
   - description: "MatchBot staging"
     url: https://matchbot-staging.thebiggivetest.org.uk/v1
+  - description: "Salesforce Full (staging)"
+    url: https://thebiggive--full.sandbox.my.salesforce-sites.com/donations/services/apexrest/v2.0
 info:
-  description: The Big Give Donations public API
-  version: 1.0.10
+  description: |
+    The Big Give Donations APIs.
+
+    This contract aims to cover all endpoints except
+    `/ping` and integration webhooks. Most webhooks' structures are defined by Stripe
+    with us choosing just the endpoint.
+
+    As Salesforce & MatchBot share a Donation schema, endpoints for both are documented here
+    together; however operation `salesforceUpsert` is implemented only in Salesforce
+    and all others are implemented only in MatchBot.
+  version: 1.0.12
   title: Donations
   contact:
     email: tech@thebiggive.org.uk
   license:
     name: Apache 2.0
     url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
 paths:
-  /donations:
+  /people/{personId}/donations:
+    parameters:
+      - name: personId
+        in: path
+        schema:
+          type: string
+          format: uuid
+        description: Identifier of the donor
+        required: true
+
     post:
-      summary: Creates a donation
-      operationId: addDonation
-      description: Initiates a donation with pending status
+      operationId: createDonation
+      summary: Create a donation on behalf of a specified person
+      tags: [Core donation journey]
+      responses:
+        '201':
+          description: Donation created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DonationCreatedResponse'
+        '400':
+          description: Invalid donation properties
       requestBody:
         content:
           application/json:
             schema:
               $ref: '#/components/schemas/Donation'
-        description: Donation to create
+
+    get:
+      operationId: listDonations
+      summary: List existing complete donations made by the person
```

Comment:
do we need to document that it looks like the implementation actually ignores the person ID in the URL and only uses the ID in the JWT header sent with the request? I think this is actually the same for createDonation, and I think I considered not having the person ID in the URL when creating this route for that reason.

Reply:
I think the token verifier & middleware do require a match don't they @bdsl? If I'm reading it correctly then a mismatch gives you
> JWT error: Not authorised for person ID ...

It's true that which person the request is for is over-specified but I think it's worth it to have clearer & more consistent URIs when looking at request logs etc.